tests/integration/_experimental/dataset/test_evaluation.py

```diff
@@ -55,188 +52,169 @@ def get_df_dp(n: int = 20) -> pd.DataFrame:
         model_name,
-        infer=get_infer_func(df_inf, inf_columns),
-        eval=get_eval_func(df_mtr, mtr_columns),
+        [(eval_config_1, df_result_1), (eval_config_2, df_result_2)],
+        on="user_dp_id",
     )
 
-    df_by_eval = fetch_evaluation_results(dataset_name, model_name)
-    eval_cfg, df_datapoints, df_inferences, df_metrics = df_by_eval[0]
-    expected_df_dp = df_dp[13:20].reset_index(drop=True)
-    expected_df_inf = pd.concat([df_inf[13:20]], axis=0).reset_index(
-        drop=True,
-    )
-    expected_df_mtr = pd.concat([df_mtr[13:20]], axis=0).reset_index(
-        drop=True,
-    )
-    assert len(df_by_eval) == 1
-    assert eval_cfg is None
-    _assert_frame_equal(df_datapoints, expected_df_dp, dp_columns)
-    _assert_frame_equal(df_inferences, expected_df_inf, inf_columns)
-    _assert_frame_equal(df_metrics, expected_df_mtr, mtr_columns)
-
-    # add 3 new datapoints, then we should have missing inference and metrics in the db records
-    register_dataset(dataset_name, df_dp[10:20][dp_columns])
-
-    df_by_eval = fetch_evaluation_results(dataset_name, model_name)
-    eval_cfg, df_datapoints, df_inferences, df_metrics = df_by_eval[0]
-    expected_df_dp = pd.concat([df_dp[13:20], df_dp[10:13]]).reset_index(drop=True)
-    expected_df_inf = pd.concat(
-        [df_inf[13:20], pd.DataFrame({"softmax_bitmap": [np.nan] * 3})],
-        axis=0,
-    ).reset_index(drop=True)
-    expected_df_mtr = pd.concat([df_mtr[13:20], pd.DataFrame({"score": [np.nan] * 3})], axis=0).reset_index(
-        drop=True,
-    )
-    assert len(df_by_eval) == 1
-    assert eval_cfg is None
-    _assert_frame_equal(df_datapoints, expected_df_dp, dp_columns)
-    _assert_frame_equal(df_inferences, expected_df_inf, inf_columns)
-    _assert_frame_equal(df_metrics, expected_df_mtr, mtr_columns)
+    fetched_df_dp, df_results_by_eval = fetch_results(dataset_name, model_name)
+    assert len(df_results_by_eval) == 2
+    expected_df_dp = df_dp[3:10].reset_index(drop=True)
+    _assert_frame_equal(fetched_df_dp, expected_df_dp, dp_columns)
+
+    df_results_by_eval = sorted(df_results_by_eval, key=lambda x: x[0].get("threshold"))
+    fetched_eval_config_1, fetched_df_result_1 = df_results_by_eval[0]
+    fetched_eval_config_2, fetched_df_result_2 = df_results_by_eval[1]
+    assert fetched_eval_config_1 == eval_config_1
+    assert fetched_eval_config_2 == eval_config_2
+    expected_df_result_1 = df_result_1[3:10].reset_index(drop=True)
+    expected_df_result_2 = df_result_2[3:10].reset_index(drop=True)
+    _assert_frame_equal(fetched_df_result_1, expected_df_result_1, result_columns_1)
+    _assert_frame_equal(fetched_df_result_2, expected_df_result_2, result_columns_2)
+
+
+def test__test__multiple_eval_configs__duplicate() -> None:
+    dataset_name = with_test_prefix(f"{__file__}::test__test__multiple_eval_configs__duplicate")
+    model_name = with_test_prefix(f"{__file__}::test__test__multiple_eval_configs__duplicate")
```

Comment:
need to prevent this on sdk-server side as well

Reply:
ack

Reply:
cut the [ticket](https://linear.app/kolena/issue/KOL-3742)